spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/config/AbstractJdbcConfiguration.java

```diff
@@ -63,10 +63,7 @@
 	public JdbcMappingContext jdbcMappingContext(Optional<NamingStrategy> namingStrategy,
 			JdbcCustomConversions customConversions) {
 
-		JdbcMappingContext mappingContext = new JdbcMappingContext(namingStrategy.orElse(NamingStrategy.INSTANCE));
-		mappingContext.setSimpleTypeHolder(customConversions.getSimpleTypeHolder());
-
-		return mappingContext;
+		return new JdbcMappingContext(namingStrategy.orElse(NamingStrategy.INSTANCE));
```

Comment:
`CustomConversions` must be propagated here, otherwise, we lose external configuration.

Reply:
Duh!

Fixed that and added a test.